### ISSUE TITLE
Windows Console: cterm colour names not working after setting termguicolors

### ISSUE THREAD

**Issue by adah1972** [vim/vim#6775]
**cterm colour names do not work in Windows with `termguicolors`**

According to `:help cterm-colors`, it seems that I should be able to write `hi Normal ctermfg=DarkCyan` to get DarkCyan on both xterm and Windows console. However, I got Yellow instead, if `termguicolors` is set under the Windows Command Prompt.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Run `vim --clean`
2. Open any file, or open the help
3. Type `:hi Normal ctermfg=DarkCyan`
4. The text colour is the correct DarkCyan
5. Type `:set termguicolors`
6. The text colour becomes Yellow

**Expected behavior**
Text colour is always the specified one, regardless of `termguicolors`.

**Environment (please complete the following information):**
 - Vim version 8.2.1459
 - OS: Windows 10 1909
 - Terminal: Command Prompt

**Comment by chrisbra** [comment]
does windows terminal actually support `termguicolors` or are you talking about the new Windows terminal?

**Comment by adah1972** [comment]
The Windows command prompt console does support `termguicolors`. The vim help writes:

```
	For Win32 console, Windows 10 version 1703 (Creators Update) or later
	is required. Use this check to find out: >
		if has('vcon')
<	This requires Vim to be built with the |+vtp| feature.
```

The options works, but it has strange side-effects. Or maybe not that strange, as the relationship between the wrong color and the *correct* color is obviously those between NR-16 and NR-8 in `:help cterm-colors`.

**Comment by k-takata** [comment]
Looks like a duplicate of #1740.

**Comment by adah1972** [comment]
The problem seems solved, as the colour switch between steps 5 and 6 ceases to happen now in the current Vim 9 build.

However, a similar colour switch will happen if one turns *off* `termguicolors`. I guess this is not as significant a problem. So it may be closed.

**Comment by chrisbra** [comment]
Yes, there is a similar issue after turning off `termguicolors`, see #9498
